packages/roosterjs-editor-plugins/lib/plugins/ImageSelection/ImageSelection.ts

```diff
@@ -37,35 +49,52 @@ export default class ImageSelection implements EditorPlugin {
     dispose() {
         this.editor.select(null);
         this.editor = null;
-        this.imageId = null;
     }
 
     onPluginEvent(event: PluginEvent) {
         if (this.editor) {
             switch (event.eventType) {
                 case PluginEventType.EnteredShadowEdit:
+                case PluginEventType.LeavingShadowEdit:
                     const selection = this.editor.getSelectionRangeEx();
                     if (selection.type == SelectionRangeTypes.ImageSelection) {
                         this.editor.select(selection.image);
                     }
                     break;
-                case PluginEventType.LeavingShadowEdit:
-                    if (this.imageId) {
-                        const images = this.editor.queryElements(
-                            '#' + this.imageId
-                        ) as HTMLImageElement[];
-                        if (images.length == 1) {
-                            const image = images[0];
-                            this.editor.select(image);
-                        }
-                        this.imageId = null;
-                    }
-                    break;
+
                 case PluginEventType.MouseDown:
                     const target = event.rawEvent.target;
                     if (safeInstanceOf(target, 'HTMLImageElement')) {
                         this.editor.select(target);
-                        this.imageId = target.id;
+                    }
+                    break;
+                case PluginEventType.KeyDown:
+                    const key = event.rawEvent.key;
+                    const keyDownSelection = this.editor.getSelectionRangeEx();
+                    if (keyDownSelection.type === SelectionRangeTypes.ImageSelection) {
+                        if (
+                            key === Escape ||
+                            key === PageDown ||
+                            key === PageUp ||
```

Comment:
also need to handle up, down, left, right

Reply:
For up, down, left, right, the range of the image is selected and the browser handles the action

Reply:
where is this logic?

Reply:
In the else case, the ImageSelectionRange store the range of the image, so when we select the range inside this object we do normal selection. 
![arrowKeys](https://user-images.githubusercontent.com/87443959/193043017-fe2282d8-0b2f-4ac4-a60a-514bc0573ede.gif)